connectors/byoei.py

```diff
@@ -325,34 +329,16 @@ def __init__(self, elastic_config):
         super().__init__(elastic_config)
         self.loop = asyncio.get_event_loop()
 
-    async def prepare_index(
-        self, index, *, docs=None, settings=None, mappings=None, delete_first=False
-    ):
+    async def prepare_index(self, index, *, mappings=None):
         """Creates the index, given a mapping if it does not exists."""
-        if index.startswith("."):
-            expand_wildcards = "hidden"
-        else:
-            expand_wildcards = "open"
+        expand_wildcards = ElasticServer._get_expand_wildcards(index)
```

Comment:
why not `self._get_expand_wildcards(index)` ?

Reply:
`_get_expand_wildcards` is a pure function that does not need access to `self`.

Reply:
that's what you did, you made it a method. you have two options: make it a method (and you need self) or make it a pure function:

```
def _get_expand_wildcards(index):
  XXX


class ElasticServer:
    def use_it(self): 
         _get_expand_wildcards(some_index)
```


OR

```
class ElasticServer:
    def _get_expand_wildcards(self, index):
      XXX

    def use_it(self): 
         self._get_expand_wildcards(some_index)
```

Reply:
I see! I ended up dropping this function fully after refactoring :)